application/controllers/Home.php

```diff
@@ -36,29 +37,63 @@ public function index()
 	}
 
 	public function do_upload()
-	{
+	{ 
 		$config['upload_path']          = './uploads/';
-		$config['allowed_types']        = 'gif|jpg|png';
+		$config['allowed_types']        = 'pdf|jpg|png';
 		$config['max_size']             = 1024;
 		// $config['max_width']            = 2048;
 		// $config['max_height']           = 2048;
-		$data['title'] = 'Home';
+		$data['tab'] = 'Oops!';
+		$data['title'] = 'Ha habido un error';
 
 		$this->load->library('upload', $config);
 
-		if ( ! $this->upload->do_upload('userfile'))
+		$filesError = array(0,0); //0 = Error, 1 = Success
+
+		if ( ! $this->upload->do_upload('userfileplan'))
+		{
+		}
+		else
 		{
-			$error = array('error' => $this->upload->display_errors());
+			 $filesError[0] = "1";
+		}
 
-			$this->load->view('header', $data);
-			$this->load->view('home', $error);
-			$this->load->view('footer');
+		if ( ! $this->upload->do_upload('userfileform'))
+		{
 		}
 		else
 		{
-			$data = array('upload_data' => $this->upload->data());
+			 $filesError[1] = "1";
+		}
+
 
+		if(array_sum($filesError) == 2)
+		{
+			$data = array('upload_data' => $this->upload->data());
```

Comment:
Codigo repetido, mucho cuidado al repetir codigo en programacion, siempre es mejor crear una función y reutilizarla si es que es necesario. Chequea DRY en programación que significa Do not repeat yourself.

Reply:
Enterado